src/fal/cli/dbt_runner.py

```diff
@@ -166,3 +165,76 @@ def _create_fal_result_file(target_path: str, run_index: int):
         shutil.copy(
             fal_run_result, os.path.join(target_path, f"fal_results_{run_index}.json")
         )
+
+
+# This is the Python implementation of the `dbt_run()` function, in which
+# we directly use dbt-core as a Python library. We don't run it directly
+# but rather use 'multiprocessing' to run it in a real system Process to
+# imitate the existing behavior of `dbt_run()` (in terms of performance).
+
+# dbt-core is currently using the spawn as its mulitprocessing context
+# so we'll mirror it.
+multiprocessing.set_start_method("spawn", force=True)
+
+
+def _dbt_run_through_python(args: List[str], target_path: str, run_index: int):
+    from dbt.main import handle_and_check
+
+    try:
+        run_results, success = handle_and_check(args)
+    except BaseException as exc:
```

Comment:
Why not `Exception`?

Reply:
If there is anything that raises a `SystemExit()` (which inherits from `BaseException`), then fal will quit as well. So we are capturing that too.

Reply:
Maybe something to consider in other places later! Thanks.